<launch>
  <!-- Sử dụng thời gian mô phỏng -->
  <param name="/use_sim_time" value="true" />

  <!-- Khởi chạy Gazebo với một thế giới trống -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch" />

  <!-- Static transform từ base_link đến base_footprint -->
  <node
    name="tf_footprint_base"
    pkg="tf"
    type="static_transform_publisher"
    args="0 0 0 0 0 0 base_link base_footprint 40" />

  <!-- Spawn robot model vào Gazebo -->
  <node
    name="spawn_model"
    pkg="gazebo_ros"
    type="spawn_model"
    args="-file $(find myomnicar)/urdf/myomnicar.urdf -urdf -model myomnicar"
    output="screen" />

  <!-- Load controller parameters -->
  <rosparam file="$(find myomnicar)/config/controllers.yaml" command="load" />

  <!-- Khởi chạy ROS Control -->
  <node
    name="controller_spawner"
    pkg="controller_manager"
    type="spawner"
    args="LeFro_joint_velocity_controller LeBa_joint_velocity_controller RiFro_joint_velocity_controller RiBa_joint_velocity_controller link1_rot_position_controller link2_pris_position_controller joint_state_controller"
    output="screen" />

  <!-- Node điều khiển omni 4 bánh -->
  <node
    name="omni_keyboard_control"
    pkg="myomnicar"
    type="omni_keyboard_control.py"
    output="screen" />

 
  <!-- Node chạy các service hiển thị giá trị sensor -->
  <node
    name="sensor_services"
    pkg="myomnicar"
    type="sensor_srv.py"
    output="screen" />
</launch>